### PR TITLE
test: setup jest to run API/notification tests in isolation

### PR DESCRIPTION
**Pull request by johnrees**
With these foundations I'll be able to move forward and develop notification features quickly and ensure that future changes won't break existing functionality.

I've written more tests and functionality for the notifier scripts but I would like to add them in a separate PR after this one if possible in order to keep things simple to review and hopefully merge.

## Notes

* `jsdom` breaks the borsh struct parsing, so the notifier tests have to run in the `node` environment. Not entirely sure what's happening here yet, but it's an interesting discovery! It'll kind of work with `jsdom`, but returns empty strings for some fields like proposal `name`

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -6,10 +6,10 @@ const Index = () => {
   useEffect(() => {
     const { REALM } = process.env
     const mainUrl = REALM ? `/dao/${REALM}` : '/realms'
-    router.push(mainUrl)
+    router.replace(mainUrl)
```
Comment (johnrees) [inline pages/index.tsx:9]: switched to replace so it that it doesn't add an extra page in the browser history

File: test/setup.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import '@testing-library/jest-dom/extend-expect'
```
Comment (johnrees) [inline test/setup.js:1]: recommended on https://nextjs.org/docs/testing

File: test/pages/index.test.tsx [hunk 3]
```diff
@@ -1,10 +1,39 @@
...
+    render(<Home />)
+
+    await waitFor(() => {
+      expect(singletonRouter).toMatchObject({
+        pathname: '/dao/MNGO',
+      })
+    })
+  })
+
+  test.todo(`/realms when process.env.REALM is not a valid realm symbol`)
```
Comment (johnrees) [inline test/pages/index.test.tsx:38]: will address this in a separate PR, just wanted to have some tests which checked that everything loaded ok

File: next.config.js [hunk 4]
```diff
@@ -8,7 +8,6 @@ const withTM = require('next-transpile-modules')([
 ])
 
 module.exports = withTM({
-  target: 'serverless',
```
Comment (johnrees) [inline next.config.js:11]: next recommended removing this when i ran `yarn test` https://nextjs.org/docs/messages/deprecated-target-config

File: test/models/registry/api.test.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+test('getCertifiedRealmInfo', async () => {
+  const mango = realms.find(({ symbol }) => symbol === 'MNGO')!
+
+  const realmInfo = await getCertifiedRealmInfo(
+    mango.symbol,
+    getConnectionContext('mainnet')
+  )
+
+  expect(realmInfo!.realmId.toBase58()).toEqual(mango.realmId)
+})
```
Comment (johnrees) [inline test/models/registry/api.test.ts:14]: I will be replacing this test with more useful ones but this is just to ensure that I can run tests without loading any frontend/next-specific code

**Comment by johnrees** [comment]
sorry about all the force-pushed & vercel deployment noise here, I'll try to avoid that in future PRs

File: next.config.js (hunk 4 above)
Comment (SebastianBor) [reply next.config.js:11]: :+1:

File: pages/index.tsx (hunk 1 above)
Comment (SebastianBor) [reply pages/index.tsx:9]: :+1:

File: test/models/registry/api.test.ts (hunk 5 above)
Comment (SebastianBor) [reply test/models/registry/api.test.ts:14]: :+1:

File: test/pages/index.test.tsx (hunk 3 above)
Comment (SebastianBor) [reply test/pages/index.test.tsx:38]: :+1:

File: test/setup.js (hunk 2 above)
Comment (SebastianBor) [reply test/setup.js:1]: :+1:

**Review by SebastianBor — COMMENTED** [review]
Superb

**Pull request merged**